atom/browser/atom_javascript_dialog_manager.cc

```diff
@@ -30,6 +31,15 @@ void AtomJavaScriptDialogManager::RunJavaScriptDialog(
     const base::string16& default_prompt_text,
     const DialogClosedCallback& callback,
     bool* did_suppress_message) {
+  std::string origin = origin_url.GetOrigin().spec();
+  if (origin_counts_.find(origin) == origin_counts_.end()) {
+    origin_counts_[origin] = 0;
+  }
+
+  if (origin_counts_[origin] == -1) {
```

Comment:
Not in love with the use of `-1` as a magic number here.

It's awkward to mix the `int` to be both a count and a flag, but I see the reasoning because once the flag is set, we never care about the count again. But let's at least give the the flag a symbolic name, e.g. at the top of the file have

```c++
namespace {
  constexpr int USER_WANTS_NO_MORE_DIALOGS = -1;
}
```

and then use `USER_WANTS_NO_MORE_DIALOGS` here and in `OnMessageBoxCallback`